src/develop/imageop.c

```diff
@@ -1021,6 +1022,20 @@ static void dt_iop_gui_off_callback(GtkToggleButton *togglebutton, gpointer user
       if(module->dev->proxy.chroma_adaptation == module)
         module->dev->proxy.chroma_adaptation = NULL;
 
+      // if a mask is visualized, reset the mask visualization and buttons
+      if(module->request_mask_display)
+      {
+        // reset the mask visualization and buttons
+        module->request_mask_display = DT_DEV_PIXELPIPE_DISPLAY_NONE;
+        module->suppress_mask = 0;
```

Comment:
well in you 5th points this path gets executed and you set "suppress_mask = 0" which means really to suppress the mask and behave as if having a uniform mask which is what is happening. So I fear that this does not make the bug appear but create just a very similar bug :)

Reply:
Well, no.
`module->suppress_mask = 0` means no suppression.
Module initialization, lose focus etc. all reset `module->suppress_mask = 0`.
Besides that, you get exactly the same issue if you comment that line

Reply:
Forget this PR, I found a way to reproduce without it, see my last comment in #8528

Reply:
> Well, no.
> `module->suppress_mask = 0` means no suppression.
> Module initialization, lose focus etc. all reset `module->suppress_mask = 0`.
> Besides that, you get exactly the same issue if you comment that line

Stupid me :) I'd my brain broken by a double negation :)

Reply:
Closing this one now that we have a reproducer.